### PR TITLE
Force POSIX format for destination (device) path operations on Windows.

### PR DESCRIPTION
**Pull request by srlowe**
Fixes issues with livepush changes not getting applied properly from Windows.

Resolves: #55
Change-type: patch
Signed-off-by: Scott Lowe <scott@balena.io>

### CODE REVIEW

File: lib/stage-copy.ts [hunk 1]
```diff
@@ -147,6 +147,7 @@ export function resolveFileDestination(
 	dest: string,
 	pathInTar: string,
 ) {
+	// TODO: source.split probably needs to use native separator (i.e. path.sep), but needs investigating.
 	const sourceParts = source.split('/').filter(p => p !== '');
```
Comment (srlowe) [inline lib/stage-copy.ts]: As per comment, this line probably needs changing.  However I haven't yet been able to trigger this codepath to test.

File: lib/container.ts [hunk 2]
```diff
@@ -311,7 +311,6 @@ export class Container extends (EventEmitter as {
 					// check the string itself. Any path ending with / is quite clearly
 					// a directory, and it means that we can avoid making another docker
 					// call
-					// TODO: Think about how this is affected when using windows
```
Comment (srlowe) [inline lib/container.ts]: I think this TODO no longer required, as `copy.dest` should now always be POSIX compliant.

**Comment by srlowe** [comment]
@balena-ci retest

**Comment by srlowe** [comment]
@balena-ci retest

File: lib/container.ts (hunk 2 above)
Comment (CameronDiver) [reply lib/container.ts]: Yeah, let's remove this

**Comment by srlowe** [comment]
This has been tested manually on windows, by applying the changes to the livepush files in the `node_modules` folder of the CLI (then executed using the `/bin/balena-dev` on-the-fly compilation).

File additions, modifications and removals were tested with multiple nested folders.   Livepush now generally works with Windows, but the following problems were encountered:
 
 1. https://github.com/balena-io-modules/livepush/issues/59 
 2. https://github.com/balena-io-modules/livepush/issues/57
 3. https://github.com/balena-io-modules/livepush/issues/60

All of these appear to be separate issues.  No. 3 is the most concerning, since it means that files cannot be properly deleted using livepush on windows - however it also seems to be a separate issue (not related to incorrectly formatted file paths).

**Pull request merged**